pulp_2to3_migration/app/plugin/rpm/pulp_2to3_models.py

```diff
@@ -93,32 +104,17 @@ async def pre_migrate_content_detail(cls, content_batch):
              content_batch(list of Pulp2Content): pre-migrated generic data for Pulp 2 content.
 
         """
-
         pulp2_id_obj_map = {pulp2content.pulp2_id: pulp2content for pulp2content in content_batch}
         pulp2_ids = pulp2_id_obj_map.keys()
-        pulp2_content_batch = pulp2_models.RPM.objects.filter(id__in=pulp2_ids).as_pymongo().only(
-            'name',
-            'epoch',
-            'version',
-            'release',
-            'arch',
-            'checksum',
-            'checksumtype',
-            'repodata',
-            'is_modular',
-            'size',
-            'filename',
-            'pk',
-        )
-        import gzip
-
+        if cls.pulp2_type == 'rpm':
+            pulp2_content_batch = pulp2_models.RPM.objects.filter(
+                id__in=pulp2_ids).as_pymongo().only(*RPM_UNIT_FIELDS)
+        elif cls.pulp2_type == 'srpm':
+            pulp2_content_batch = pulp2_models.SRPM.objects.filter(
+                id__in=pulp2_ids).as_pymongo().only(*SRPM_UNIT_FIELDS)
         pulp2rpm_to_save = []
         for rpm in pulp2_content_batch:
-            compressed_repodata = rpm['repodata']
-            decompressed_repodata = {}
-            for name, gzipped_data in compressed_repodata.items():
-                decompressed_repodata[name] = gzip.zlib.decompress(
-                    bytearray(gzipped_data)).decode()
+            decompressed_repodata = decompress_repodata(rpm['repodata'])
             rpm['repodata'] = decompressed_repodata
```

Comment:
I think this should be enough `rpm['repodata'] = decompress_repodata(rpm['repodata'])`